### ISSUE TITLE
Resolve empty WAL file issue

### ISSUE THREAD

**Issue by yatharthranjan** [RADAR-base/RADAR-HDFS-Sink-Connector#16]
Resolve the issue mentioned at https://github.com/confluentinc/kafka-connect-hdfs/issues/53

**Comment by yatharthranjan** [comment]
should be fixed in confluent version 5.0.x by this PR https://github.com/confluentinc/kafka-connect-hdfs/pull/348

**Comment by rupeshmore85** [comment]
Is this issue fixed in 5.0.0 ? I see this error appearing in confluent 5.0.0 as well. Thanks.
```
[2018-09-16 22:44:21,946] INFO Successfully acquired lease for hdfs://hadoop-das//home/scribe/kafka_connect/logs/scribeLogBackend/10/log (io.confluent.connect.hdfs.wal.FSWAL)
[2018-09-16 22:44:21,955] ERROR Error applying WAL file: hdfs://hadoop-das//home/scribe/kafka_connect/logs/scribeLogBackend/10/log, {} (io.confluent.connect.hdfs.wal.FSWAL)
java.io.IOException: File is corrupt!
    at io.confluent.connect.hdfs.wal.WALFile$Reader.readRecordLength(WALFile.java:749)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.next(WALFile.java:817)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.next(WALFile.java:775)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.next(WALFile.java:801)
    at io.confluent.connect.hdfs.wal.FSWAL.apply(FSWAL.java:112)
    at io.confluent.connect.hdfs.TopicPartitionWriter.applyWAL(TopicPartitionWriter.java:601)
    at io.confluent.connect.hdfs.TopicPartitionWriter.recover(TopicPartitionWriter.java:256)
    at io.confluent.connect.hdfs.TopicPartitionWriter.write(TopicPartitionWriter.java:321)
    at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:375)
    at io.confluent.connect.hdfs.HdfsSinkTask.put(HdfsSinkTask.java:109)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:564)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:322)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:225)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:193)
    at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:175)
    at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:219)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
[2018-09-16 22:44:21,990] ERROR Recovery failed at state RECOVERY_PARTITION_PAUSED (io.confluent.connect.hdfs.TopicPartitionWriter)
org.apache.kafka.connect.errors.DataException: java.io.IOException: File is corrupt!
    at io.confluent.connect.hdfs.wal.FSWAL.apply(FSWAL.java:133)
    at io.confluent.connect.hdfs.TopicPartitionWriter.applyWAL(TopicPartitionWriter.java:601)
    at io.confluent.connect.hdfs.TopicPartitionWriter.recover(TopicPartitionWriter.java:256)
    at io.confluent.connect.hdfs.TopicPartitionWriter.write(TopicPartitionWriter.java:321)
    at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:375)
    at io.confluent.connect.hdfs.HdfsSinkTask.put(HdfsSinkTask.java:109)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:564)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:322)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:225)
    at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:193)
    at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:175)
    at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:219)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: File is corrupt!
    at io.confluent.connect.hdfs.wal.WALFile$Reader.readRecordLength(WALFile.java:749)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.next(WALFile.java:817)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.next(WALFile.java:775)
    at io.confluent.connect.hdfs.wal.WALFile$Reader.next(WALFile.java:801)
    at io.confluent.connect.hdfs.wal.FSWAL.apply(FSWAL.java:112)
    ... 16 more
```

**Comment by rupeshmore85** [comment]
@yatharthranjan : Can you please take at the above error when you can?

**Comment by yatharthranjan** [comment]
Hi rupesh, we are still using v0.4.1 in production so can't comment on if this is fixed in v0.5.0.
But i do know how to get rid of this error. Its basically because of empty/corrupt log files an deleting them usually fixes this. I think the empty file issue was fixed in v0.5.0 but the corrupt file still throws this error.
If not using the connector in distributed mode, its safe to delete everything in the logs directory. If using in distributed mode then just delete the corrupt file. Make sure to backup before deleting. Hope this helps.

**Comment by rupeshmore85** [comment]
@yatharthranjan We are running kafka connect hdfs in distributed fashion within kubernetes for more than 40 connectors, so that its unlikely to delete those corrupt log files when connectors restart. 
Shall i log a separate bug for this ? 
What's the downside of deleting these WAL log files every 10 mins ? I mean what happens to active connector tasks ?
